Microsoft.Alm.Authentication/Src/Settings.cs

```diff
@@ -85,6 +85,11 @@ public string[] GetCommandLineArgs()
             {
                 if (key is string name && variables[key] is string value)
                 {
+                    // avoid trying to add duplicates, e.g. different case names, last entry wins
+                    if (result.ContainsKey(name))
+                    {
+                        result.Remove(name);
+                    }
                     result.Add(name, value);
```

Comment:
You can just replace this with:

```suggestion
                    result[name] = value;
```

..and that will also do the same thing.